src/vm/instruction.test.ts

```diff
@@ -15,6 +21,54 @@ import { SignedInteger16 } from 'primitives/int';
 
 describe('Instruction', () => {
   describe('decodeInstruction', () => {
+    test('should fail with HighBitSetError', () => {
```

Comment:
did we try and incorporate tests from cairovmgo, rs, nethermind?

Reply:
(if we do and i think we should take one from each for each test, good to credit as well)

Reply:
erratum!! i didn't read 5 lines down! 
very nice tests and nice porting of existing tests